src/software/geom/linear_spline2d_test.cpp

```diff
@@ -117,3 +117,46 @@ TEST(LinearSpline2dTest, test_get_knot_parametrization_values)
 
     EXPECT_EQ(std::vector<double>({0, 0.5, 1}), s.getKnotParametrizationValues());
 }
+
+TEST(LinearSpline2dTest, test_get_spline_segments)
+{
+    // linear spline with three points & two segments
+    {
+        LinearSpline2d spline({Point(1, 2), Point(2, 3), Point(0, -1)});
+        std::vector<SplineSegment2d> spline_segments = spline.getSplineSegments();
+        EXPECT_EQ(spline_segments.size(), 2);
+
+        Polynomial1d segment1_x({{0, 1}, {1.0 / 2.0, 2}});
```

Comment:
Some comments about what the polynomial part of the test is doing would be helpful